core/base/document.php

```diff
@@ -1499,4 +1523,48 @@ private function add_handle_revisions_changed_filter() {
 	private function remove_handle_revisions_changed_filter() {
 		remove_filter( 'wp_save_post_revision_post_has_changed', [ $this, 'handle_revisions_changed' ] );
 	}
+
+	private function get_runtime_data_updaters() {
+		$runtime_data_updaters = [];
+
+		foreach ( $this->data_updaters as $data_updater ) {
+			if ( $data_updater->is_update_needed() ) {
+				$runtime_data_updaters[] = $data_updater;
+			}
+		}
+
+		return $runtime_data_updaters;
+	}
+
+	private function handle_page_elements( $elements, $data_updaters, $event ) {
+		$unique_page_widgets = [];
+
+		Plugin::$instance->db->iterate_data( $elements, function( $element_data ) use ( &$unique_page_widgets, &$data_updaters ) {
+			$widget_name = isset( $element_data['widgetType'] ) ? $element_data['widgetType'] : '';
+
+			$element = Plugin::$instance->elements_manager->create_element_instance( $element_data );
+
+			if ( $widget_name && ! in_array( $widget_name, $unique_page_widgets, true ) ) {
+				$unique_page_widgets[] = $widget_name;
+
+				foreach ( $data_updaters as $data_updater ) {
+					$data_updater->update_unique_widget( $element );
+				}
+			}
+
+			foreach ( $data_updaters as $data_updater ) {
+				$data_updater->update_element( $element );
+			}
+
+			return $element_data;
+		} );
+
+		foreach ( $data_updaters as $data_updater ) {
+			$data_updater->after_elements_iteration( $event );
```

Comment:
Instead of that, allow settings iteration mode

Reply:
Done.